utils/undoredo/undoredo.ts

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+//
+// This code is heavily based on the undoredu.js library from
+// https://github.com/iMrDJAi/UndoRedo.js with MIT license and copyrighted by
```

Comment:
I think this probably needs to be added to the NOTICE.txt even if we're using a modified version of the library in-line

Reply:
Also, what are we changing here? Maybe we could make a note of it in case we ever need to update from the upstream library.

Reply:
We are changing a big chunk of the code, I leave that message there to honor the license, I can add this to the NOTICE.txt.

Reply:
Done